mpqp/noise/noise_model.py

```diff
@@ -93,10 +100,52 @@ def connections(self) -> set[int]:
         return set(self.targets)
 
     @abstractmethod
-    def to_kraus_representation(self) -> KrausRepresentation:
-        """3M-TODO: to be implemented"""
+    def to_kraus_operators(self) -> list[npt.NDArray[np.complex64]]:
+        r"""Noise models can be represented by Kraus operators. They represent how the
+        state is affected by the noise following the formula
+
+        `\rho \leftarrow \sum_{K \in \mathcal{K}} K \rho K^\dagger`
+
+        Where `\mathcal{K}` is the set of Kraus operators corresponding to the
+        noise model and `\rho` is the state (as a density matrix).
+
+        Returns:
+            The Kraus operators of the noise. Note that it is not a unique
+            representation.
+        """
         pass
 
+    def to_adjusted_kraus_operators(
+        self, targets: set[int], size: int
+    ) -> list[npt.NDArray[np.complex64]]:
+        r"""In some cases, you may prefer the Kraus operators to match the size
+        of your circuit, and the targets involved. In particular, the targets of
+        the noise application may not match the noise targets, because the noise
+        targets signifies all the qubits that the noise is applicable on, but if
+        the noise happens at a gate execution, it would only actually impact the
+        targets qubits of the gate.
+
+        Note:
+            This generic method considers that the default Kraus operators of
+            the noise are for one qubit noises. If this is not the case, this
+            method should be overloaded in the corresponding class.
```

Comment:
hmmm, to make sure I understand, let me take an example:
for a non dimensional noise, the operators would be p1II, p2IK, p3KI and p4KK,
while for a dimensional it would just be p1II and p2K2

Reply:
since this is a bit complex it will come in a future PR so that we can merge this one